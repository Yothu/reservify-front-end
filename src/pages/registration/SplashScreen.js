import style from './SplashScreen.module.css';
import TopMenu from '../../components/SplashScreen/TopMenu/TopMenu';
import MainImage from '../../components/SplashScreen/MainImage/MainImage';
import AuthButtons from '../../components/SplashScreen/AuthButtons/AuthButtons';

const SplashScreen = () => (
  <section className={style['page-container']}>
    <div className={style['logout-menu']}>
      <TopMenu />
      <MainImage />
      <AuthButtons />
    </div>
  </section>
);

export default SplashScreen;
