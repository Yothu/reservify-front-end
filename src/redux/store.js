import { configureStore } from '@reduxjs/toolkit';
import hotelReducer from './hotels/hotel-slice';
import authReducer from './auth/auth-slice';
import reservationReducer from './reservations/reservation-slice';
import locationReducer from './locations/location-slice';

const store = configureStore({
  reducer: {
    auth: authReducer,
    hotel: hotelReducer,
    reservation: reservationReducer,
    location: locationReducer,
  },
});

export default store;
