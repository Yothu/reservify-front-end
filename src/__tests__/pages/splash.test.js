import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import SplashScreen from '../../pages/registration/SplashScreen';

describe('Login Page', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <SplashScreen />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
