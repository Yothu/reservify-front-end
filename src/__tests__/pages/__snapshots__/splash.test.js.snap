// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="page-container"
      >
        <div
          class="logout-menu"
        >
          <nav
            class="top-bar-container"
          >
            <div
              class="search-bar"
            >
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 183, 1);"
                viewBox="0 0 512 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                />
              </svg>
            </div>
          </nav>
          <div
            class="outer-hotel-container"
          >
            <div
              class="splash-background-container"
            >
              <img
                alt="hotel-background"
                class="hotel-background"
                src="room-1.jpg"
              />
              <h1
                class="app-name"
              >
                Reservify!
              </h1>
            </div>
          </div>
          <nav
            class="buttons-container"
          >
            <a
              class="link-button"
              href="/signup"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 1024 1024"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs />
                <path
                  d="M521.7 82c-152.5-.4-286.7 78.5-363.4 197.7-3.4 5.3.4 12.3 6.7 12.3h70.3c4.8 0 9.3-2.1 12.3-5.8 7-8.5 14.5-16.7 22.4-24.5 32.6-32.5 70.5-58.1 112.7-75.9 43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 32.6 32.5 58.1 70.4 76 112.5C865.7 417.8 875 464.1 875 512c0 47.9-9.4 94.2-27.8 137.8-17.8 42.1-43.4 80-76 112.5s-70.5 58.1-112.7 75.9A352.8 352.8 0 0 1 520.6 866c-47.9 0-94.3-9.4-137.9-27.8A353.84 353.84 0 0 1 270 762.3c-7.9-7.9-15.3-16.1-22.4-24.5-3-3.7-7.6-5.8-12.3-5.8H165c-6.3 0-10.2 7-6.7 12.3C234.9 863.2 368.5 942 520.6 942c236.2 0 428-190.1 430.4-425.6C953.4 277.1 761.3 82.6 521.7 82zM395.02 624v-76h-314c-4.4 0-8-3.6-8-8v-56c0-4.4 3.6-8 8-8h314v-76c0-6.7 7.8-10.5 13-6.3l141.9 112a8 8 0 0 1 0 12.6l-141.9 112c-5.2 4.1-13 .4-13-6.3z"
                />
              </svg>
              <span
                class="link-text"
              >
                Sign Up
              </span>
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 1024 1024"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M666.7 505.5l-246-178A8 8 0 0 0 408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
                />
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                />
              </svg>
            </a>
            <a
              class="link-button"
              href="/login"
            >
              <svg
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                style="color: white;"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"
                />
                <circle
                  cx="8.5"
                  cy="7"
                  r="4"
                />
                <line
                  x1="20"
                  x2="20"
                  y1="8"
                  y2="14"
                />
                <line
                  x1="23"
                  x2="17"
                  y1="11"
                  y2="11"
                />
              </svg>
              <span
                class="link-text"
              >
                Log In
              </span>
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 1024 1024"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M666.7 505.5l-246-178A8 8 0 0 0 408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
                />
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                />
              </svg>
            </a>
          </nav>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="page-container"
    >
      <div
        class="logout-menu"
      >
        <nav
          class="top-bar-container"
        >
          <div
            class="search-bar"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(255, 183, 1);"
              viewBox="0 0 512 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </div>
        </nav>
        <div
          class="outer-hotel-container"
        >
          <div
            class="splash-background-container"
          >
            <img
              alt="hotel-background"
              class="hotel-background"
              src="room-1.jpg"
            />
            <h1
              class="app-name"
            >
              Reservify!
            </h1>
          </div>
        </div>
        <nav
          class="buttons-container"
        >
          <a
            class="link-button"
            href="/signup"
          >
            <svg
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 1024 1024"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs />
              <path
                d="M521.7 82c-152.5-.4-286.7 78.5-363.4 197.7-3.4 5.3.4 12.3 6.7 12.3h70.3c4.8 0 9.3-2.1 12.3-5.8 7-8.5 14.5-16.7 22.4-24.5 32.6-32.5 70.5-58.1 112.7-75.9 43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 32.6 32.5 58.1 70.4 76 112.5C865.7 417.8 875 464.1 875 512c0 47.9-9.4 94.2-27.8 137.8-17.8 42.1-43.4 80-76 112.5s-70.5 58.1-112.7 75.9A352.8 352.8 0 0 1 520.6 866c-47.9 0-94.3-9.4-137.9-27.8A353.84 353.84 0 0 1 270 762.3c-7.9-7.9-15.3-16.1-22.4-24.5-3-3.7-7.6-5.8-12.3-5.8H165c-6.3 0-10.2 7-6.7 12.3C234.9 863.2 368.5 942 520.6 942c236.2 0 428-190.1 430.4-425.6C953.4 277.1 761.3 82.6 521.7 82zM395.02 624v-76h-314c-4.4 0-8-3.6-8-8v-56c0-4.4 3.6-8 8-8h314v-76c0-6.7 7.8-10.5 13-6.3l141.9 112a8 8 0 0 1 0 12.6l-141.9 112c-5.2 4.1-13 .4-13-6.3z"
              />
            </svg>
            <span
              class="link-text"
            >
              Sign Up
            </span>
            <svg
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 1024 1024"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M666.7 505.5l-246-178A8 8 0 0 0 408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
              />
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
            </svg>
          </a>
          <a
            class="link-button"
            href="/login"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="color: white;"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="8.5"
                cy="7"
                r="4"
              />
              <line
                x1="20"
                x2="20"
                y1="8"
                y2="14"
              />
              <line
                x1="23"
                x2="17"
                y1="11"
                y2="11"
              />
            </svg>
            <span
              class="link-text"
            >
              Log In
            </span>
            <svg
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 1024 1024"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M666.7 505.5l-246-178A8 8 0 0 0 408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
              />
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
            </svg>
          </a>
        </nav>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
